auth/store/mongo/device_tokens_repository.go

```diff
@@ -0,0 +1,53 @@
+package mongo
+
+import (
+	"context"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/tidepool-org/platform/devicetokens"
+	"github.com/tidepool-org/platform/errors"
+	structuredmongo "github.com/tidepool-org/platform/store/structured/mongo"
+)
+
+// deviceTokenRepo implements devicetokens.Repository, writing data to a
+// MongoDB collection.
+type deviceTokenRepo structuredmongo.Repository
+
+// Upsert will create or update the given Config.
+func (r *deviceTokenRepo) Upsert(ctx context.Context, doc *devicetokens.Document) error {
+	// The presence of UserID and TokenID should be enforced with a mongodb
+	// index, but better safe than sorry.
+	if doc.UserID == "" {
+		return errors.New("UserID may not be empty")
+	}
+	if doc.TokenKey == "" {
+		return errors.New("TokenID may not be empty")
+	}
```

Comment:
I'm updating this one, but I disagree. "TokenID is empty" isn't actionalbe. It's stating the obvious. "TokenID may not be empty" tells the user exactly what the expectation that's not being met is.

Reply:
Hmm, funny, I read "UserID may not be empty" as "UserID may (or may not) be empty".

I suppose I could see "UserID should not be empty", but again, this goes back to the convention topic above. If we wanted to globally change all cases of "is empty" with "should not be empty" then I'd be more on-board with that. However, that seems like a significant change for little gain.

Reply:
I guess it should have been "must not be empty": https://datatracker.ietf.org/doc/html/rfc2119

> However, that seems like a significant change for little gain.

Reply:
Yes, "must not be empty is even better! Particularly since it is conforming to some standard. Unfortunately, we didn't think of that way back when. 😞